src/main/java/nextstep/subway/path/domain/FareCalculator.java

```diff
@@ -0,0 +1,41 @@
+package nextstep.subway.path.domain;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import nextstep.subway.line.domain.Distance;
+import nextstep.subway.line.domain.Line;
+import nextstep.subway.policy.DiscountPolicyFactory;
+import nextstep.subway.policy.domain.Price;
+import nextstep.subway.policy.price.DistancePricePolicy;
+import nextstep.subway.policy.price.LinePricePolicy;
+import nextstep.subway.policy.price.PricePolicy;
+
+public class FareCalculator {
+    private FareCalculator() {
+        throw new IllegalAccessError("이 클래스는 유틸 클래스입니다.");
+    }
+
+    public static Price calculate(Distance distance, List<Line> lines, Integer age) {
+        Price totalFare = fareCalculate(distance, lines);
+
+        return DiscountPolicyFactory.generate(age).apply(totalFare);
+    }
+
+    public static Price calculate(Distance distance, List<Line> lines) {
+        return fareCalculate(distance, lines);
+    }
+
+    private static Price fareCalculate(Distance distance, List<Line> lines) {
+        PricePolicy linePricePolicy = new LinePricePolicy(lines);
+        PricePolicy distancePricePolicy = new DistancePricePolicy(distance);
+
+        List<PricePolicy> pricePolicys = List.of(linePricePolicy, distancePricePolicy);
+
+        Price totalFare = pricePolicys.stream()
+                                        .map(PricePolicy::apply)
+                                        .reduce((seed, result) -> seed.plus(result))
```

Comment:
이 부분도 아래처럼
```suggestion
                                        .reduce(Price::plus)
```
간단히 줄여볼 수 있습니다 🙏

Reply:
오...
초기값이 무조건 있어야되는줄 알았는데 이런 방식이 있었네요!!
감사합니다.